paddle/fluid/framework/parallel_executor.cc

```diff
@@ -254,6 +254,16 @@ ParallelExecutor::ParallelExecutor(
     member_->nccl_ctxs_.reset(new platform::NCCLContextMap(
         member_->places_, nccl_id, build_strategy.num_trainers_,
         build_strategy.trainer_id_));
+    // initialize device context's nccl comm, so that all operators
+    // can use it.
+    for (size_t dev_id = 0; dev_id < member_->places_.size(); ++dev_id) {
+      auto &nccl_ctx = member_->nccl_ctxs_->at(dev_id);
+      platform::DeviceContextPool &pool =
+          platform::DeviceContextPool::Instance();
+      auto *dev_ctx = static_cast<platform::CUDADeviceContext *>(
+          pool.Get(member_->places_[dev_id]));
+      dev_ctx->set_nccl_comm(nccl_ctx.comm());
```

Comment:
Do this means computation and NCCL communication use the common CUDA stream? I think this may cause performance degradation.

Reply:
Will use the same stream for now. There's FIXME comment in `allreduce_op.cc` which will be fixed.